### PR TITLE
Update NetworkAddonsConfig Status field with conditions

### PR DESCRIPTION
**Pull request by phoracek**
With this PR we update NetworkAddonsConfig Status field with set of
conditions. These conditions include Failing, Progressing and Ready.

Failing indicates error either on rendering/applying level of
NetworkAddonsConfig operator of an error that happened during deployment
of components's pods.

Progressing indicates on-going deployment of components.

Ready set to True indicates that all components has been successfully
deployed.

Failing Status field:

```yaml
status:
  conditions:
  - lastProbeTime: 2019-04-02T14:26:54Z
    lastTransitionTime: 2019-04-02T14:26:51Z
    message: |-
      invalid configuration:
      endPoolRange must be configured
      requested imagePullPolicy 'Alwayss' is not valid
    reason: FailedToRender
    status: "True"
    type: Failing
```

Progressing Status field:

```yaml
status:
  conditions:
  - lastProbeTime: 2019-04-02T14:28:17Z
    lastTransitionTime: 2019-04-02T14:28:17Z
    status: "False"
    type: Failing
  - lastProbeTime: 2019-04-02T14:28:44Z
    lastTransitionTime: 2019-04-02T14:28:44Z
    message: |-
      DaemonSet "linux-bridge/kube-cni-linux-bridge-plugin" is not available (awaiting 1 nodes)
      DaemonSet "sriov/kube-sriov-cni-plugin" is not available (awaiting 1 nodes)
      DaemonSet "sriov/kube-sriov-device-plugin" is not available (awaiting 1 nodes)
    reason: Deploying
    status: "True"
    type: Progressing
  - lastProbeTime: 2019-04-02T14:28:44Z
    lastTransitionTime: 2019-04-02T14:28:44Z
    message: Configuration is in process
    reason: Startup
    status: "False"
    type: Ready
```

Ready Status field:

```yaml
status:
  conditions:
  - lastProbeTime: 2019-04-02T14:28:17Z
    lastTransitionTime: 2019-04-02T14:28:17Z
    status: "False"
    type: Failing
  - lastProbeTime: 2019-04-02T14:29:27Z
    lastTransitionTime: 2019-04-02T14:29:27Z
    status: "False"
    type: Progressing
  - lastProbeTime: 2019-04-02T14:29:27Z
    lastTransitionTime: 2019-04-02T14:29:27Z
    status: "True"
    type: Ready
```

### CODE REVIEW

**Comment by phoracek** [comment]
@SchSeba @booxter please take a look. I recommend reviewing it commit by commit.

**Comment by phoracek** [comment]
Need to add monitoring of `StatefulSet` used by kubemacpool. I also think that there is an issue in monitoring progress of Multus/Linux bridge.

**Comment by phoracek** [comment]
I resolved the reporting problem. Now we need to find out whether we can use `Deployment` for kubemacpool or we need to add monitoring for `StatefulSet`.

File: pkg/controller/networkaddonsconfig/pod_controller.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	r.resources = resources
+}
+
+// Reconcile updates the NetworkAddonsConfig.Status to match the current state of the
+// watched Deployments/DaemonSets
+func (r *ReconcilePods) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	found := false
+	for _, name := range r.resources {
+		if name.Namespace == request.Namespace && name.Name == request.Name {
+			found = true
```
[the remark this answers is not in the document]
Comment (phoracek) [reply pkg/controller/networkaddonsconfig/pod_controller.go]: good catch!

**Comment by ovirt-infra** [comment]
All tests passed

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go [hunk 2]
```diff
@@ -55,17 +56,18 @@ func Add(mgr manager.Manager) error {
 }
 
 // newReconciler returns a new reconcile.Reconciler
```
Comment (booxter) [inline pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: should this comment be changed?

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go [hunk 3]
```diff
@@ -55,17 +56,18 @@ func Add(mgr manager.Manager) error {
...
+func newReconciler(mgr manager.Manager, sccIsAvailable bool) *ReconcileNetworkAddonsConfig {
 	return &ReconcileNetworkAddonsConfig{
 		client:         mgr.GetClient(),
 		scheme:         mgr.GetScheme(),
+		podReconciler:  newPodReconciler(),
 		sccIsAvailable: sccIsAvailable,
 	}
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
```
Comment (booxter) [inline pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: ditto

**Comment by booxter** [comment]
I can only applause to the clarity of the code, all the comments describing the logic, the way PR is split into commits etc. Great stuff.

**Comment by phoracek** [comment]
:blush:

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go (hunk 2 above)
Comment (phoracek) [reply pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: Done

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go (hunk 3 above)
Comment (phoracek) [reply pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: Done

**Comment by ovirt-infra** [comment]
Some tests failed

**Pull request merged**